### ISSUE TITLE
Add filtering for PSA enabled records

### ISSUE THREAD

**Issue by jeffreytigch** [itglue/powershellwrapper#75]
**Is your feature request related to a problem? Please describe.**
Currently, it's not possible to filter on PSA integrations through the Powershell Module.

**Describe the solution you'd like**
Add another filter option in the commands in ITGlue, to replicate the functionality that's in the API documentation.

**Describe alternatives you've considered**
Currently, this needs to be filtered within Powershell by pulling all of the records and filter through "Where-Object".

**Comment by ecspresso** [comment]
Are you talking about contacts or other assets as well?

**Comment by jeffreytigch** [comment]
Other assets as well, if the API is able to provide that. Would save an API call, but like I said, is easily filtered at the client side.

**Comment by ecspresso** [comment]
It looks like there are two new filter parameters, `filter[psa_id]` and `filter[psa_integration_type]`. They should do what you want (maybe you had them in mind already?).

@adrianwells @CalebAlbers I could go over all the parameters and add what is missing but `filter[psa_id]` must be used together with `filter[psa_integration_type]`. What would be the best way to implement this? 

What comes to mind is either parameter sets with `filter[psa_integration_type]` as mandatory if `filter[psa_id]` is present but this would break the index/show scheme:

```posh
function Get-ITGlueContacts {
    [CmdletBinding(DefaultParameterSetName = 'index')]
    Param (
        [Parameter(ParameterSetName = 'index')]
        [Parameter(ParameterSetName = 'show')]
        [Parameter(ParameterSetName = 'psa_id')]
        [Nullable[Int64]]$organization_id = $null,

        [Parameter(ParameterSetName = 'index')]
        [Parameter(ParameterSetName = 'psa_id')]
        [Nullable[Int64]]$filter_id = $null,

        # [...]

        [Parameter(ParameterSetName = 'index')]
        [Parameter(ParameterSetName = 'psa_id')]
        [ValidateSet('manage', 'autotask', 'tigerpaw', 'kaseya-bms', 'pulseway-psa', 'vorex')]
        [String]$filter_psa_integration_type = '',

        [Parameter(ParameterSetName = 'psa_id', Mandatory=$true)]
        [String]$filter_psa_id = '',
    )

    # [...]

    $body = @{}

    if ($PSCmdlet.ParameterSetName -eq 'index' -or $PSCmdlet.ParameterSetName -eq 'psa_id') {
    # or exlude
    # if ($PSCmdlet.ParameterSetName -ne 'show') {
        # [...]
    }

    # [...]
}
```

or an if statement:
```posh
if($filter_psa_id -and -not $filter_psa_integration_type) {
    exit 4
}
```

**Comment by jeffreytigch** [comment]
Correct, that's why I have entered the issue. The filters are there in the API endpoint, but not in the wrapper. 

Weird that it needs to work together. I created this issue based on the output - that is, filtering on the value "psa-integration" being either 'enabled' or 'disabled'. For my use case, I would rather have something like that than being able to filter through PSA's that I don't use. Although I can imagine this could be used by users that have multiple PSA's linked.

**Comment by ecspresso** [comment]
>  For my use case, I would rather have something like that than being able to filter through PSA's that I don't use.

Since the API does not support filtering on enabled, only on which type, the filtering would have to be made using a `Where-Object` which is what you are already doing.

**Comment by CalebAlbers** [comment]
I implemented this as an additional parameter set (thanks for the implementation idea @ecspresso!)

Closed via PR #113, new release to ps gallery is in the works
